`tt-transformers`: Fixed unnecessary model weights cast for `HF` models (#25436)
tenstorrent/tt-metal: models/tt_transformers/README.md

**Chunked prefill (Llama3.2-11B multimodal)**: Llama3.2-11B multimodal is currently only supported on N300 and T3000. On N300, a max prefill context length of 8k is supported, while T3000 supports a max context length of 128k.

## Memory Optimization

### HuggingFace Model Caching Control

To help manage memory usage, you can control whether the HuggingFace model is cached in memory using the `cache_hf` parameter via command line or code:

```python
# For memory-constrained environments - disable caching
model_args = ModelArgs(
    mesh_device,
    cache_hf=False,  # Reduces memory usage by not keeping HF model in memory
    max_batch_size=1,
    max_seq_len=2048
)

# Default behavior - enables caching for faster repeated access
model_args = ModelArgs(
    mesh_device,
    cache_hf=True,  # Default: cache HF model for better performance
    max_batch_size=4,
    max_seq_len=4096

> )

Reviewer: The new default behaviour (as you've correctly changed it to) is that we do **not** cache by default. So these README.md changes will need updating

Author: Sorry, totally missed updating the ReadMe! Will fix this ASAP

```

**When to disable caching (`cache_hf=False`):**
- Running on systems with limited memory (< 256GB)
- Loading large models (70B+ parameters)
- Using the model for single inference runs
- When you don't need reference model comparisons

**When to keep caching enabled (`cache_hf=True`, default):**
- Sufficient memory available

> - Multiple inference runs or comparisons needed

Reviewer: ```suggestion
- Comparisons with torch model is needed
```

> - Performance is prioritized over memory usage

Reviewer: This makes it seem like the model's performance can improve if you set `cache_hf=True`.

This should read:
```suggestion
- Minimizing test duration is prioritized over memory usage
```

- Running reference model tests

The `cache_hf` parameter affects:
- `load_state_dict()` method: Controls whether HF model is cached after loading
- `reference_transformer()` method: Controls whether to reuse cached model or load fresh

**Memory Impact:**
- Disabling caching saves approximately the full model size in memory
- For a 70B model, this can save ~140GB+ of memory usage

> - Slight performance cost as model needs to be reloaded for reference operations

Reviewer: Again, this makes it seem like setting `cache_hf=True` will improve the model's performance. But this just lowers the end to end duration of the tests.
```suggestion
- Increased test duration as model needs to be reloaded for reference operations
```
